mne/io/base.py

```diff
@@ -363,6 +368,26 @@ def __init__(self, info, preload=False,
         self._comp = None
         self._filenames = list(filenames)
         self.orig_format = orig_format
+        # Sanity check and set original units, if provided by the reader:
+        if orig_units:
```

Comment:
`orig_units` should be same length as `info['ch_names']` ? and the same `ch_names`? do you validate this somewhere?

Reply:
this is a bit tricky ... I make sure that all channels and units read from the raw file are in `orig_units`. But the MNE IO procedure changes a few things on the way, e.g.:

1. adding a stim channel
2. truncating long channel names so that they have max. 15 chars
3. adding or removing some other channels depending on the IO arguments the user provides (excluding, etc.)

So bottom line: currently I only check that each channel in `info['ch_names']` does have a corresponding key in `orig_units`, which I think is what we want, and thus it's fine.

Reply:
okay that's fine then